src/OrleansCodeGenerator/CodeGeneratorCommon.cs

```diff
@@ -0,0 +1,208 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+namespace OrleansCodeGenerator
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+
+    using Microsoft.CodeAnalysis;
+    using Microsoft.CodeAnalysis.CSharp;
+    using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+    using Orleans;
+    using Orleans.Runtime;
+
+    using GrainInterfaceData = Orleans.CodeGeneration.GrainInterfaceData;
+    using SF = Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+    
+    /// <summary>
+    /// Methods common to multiple code generators.
+    /// </summary>
+    internal static class CodeGeneratorCommon
+    {
+        /// <summary>
+        /// Returns a value indicating whether or not a class should be generated for the specified type.
+        /// </summary>
+        /// <param name="type">The type.</param>
+        /// <returns>
+        /// A value indicating whether or not a class should be generated for the specified type.
+        /// </returns>
+        public static bool ShouldGenerate(Type type)
+        {
+            return type.IsPublic && GrainInterfaceData.IsGrainInterface(type) && type != typeof(IGrain)
```

Comment:
Grain interfaces and argument types can be internal. Because we codegen today within the same assembly, it works. Will this approach break for internal types?